src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -692,23 +687,29 @@ def create(exported_python_target):
 
     interpreter = self.context.products.get_data(PythonInterpreter)
     python_dists = self.context.products.register_data(self.PYTHON_DISTS_PRODUCT, {})
+
+    setup_runner = SetupPyRunner.Factory.create(
+      scope=self,
+      interpreter=interpreter,
+      pex_file_path=os.path.join(self.workdir, self.fingerprint, 'setup-py-runner.pex')
+    )
     for exported_python_target in reversed(sort_targets(list(created.keys()))):
       setup_dir = created.get(exported_python_target)
       if setup_dir:
         if not self._run:
-          self.context.log.info('Running packager against {}'.format(setup_dir))
-          setup_runner = Packager(setup_dir, interpreter=interpreter)
-          tgz_name = os.path.basename(setup_runner.sdist())
+          self.context.log.info('Running sdist against {}'.format(setup_dir))
+          sdist = setup_runner.sdist(setup_dir)
+          tgz_name = sdist.name
           sdist_path = os.path.join(dist_dir, tgz_name)
           self.context.log.info('Writing {}'.format(sdist_path))
-          shutil.move(setup_runner.sdist(), sdist_path)
-          safe_rmtree(setup_dir)
+          shutil.move(sdist, sdist_path)
+          safe_rmtree(str(setup_dir))
           python_dists[exported_python_target] = sdist_path
         else:
           self.context.log.info('Running {} against {}'.format(self._run, setup_dir))
           split_command = safe_shlex_split(self._run)
-          setup_runner = SetupPyRunner(setup_dir, split_command, interpreter=interpreter)
-          installed = setup_runner.run()
-          if not installed:
-            raise TaskError('Install failed.')
+          try:
+            setup_runner.run_setup_command(source_dir=setup_dir, setup_command=split_command)
+          except SetupPyRunner.CommandFailure as e:
+            raise TaskError('Install failed: {}'.format(e))
```

Comment:
```suggestion
            raise TaskError(f'Install failed: {e!r}')
```

Reply:
For example re repr, it seems definitely wrong to me here. TaskError is guaranteed to be displayed directly to the user.